### ISSUE TITLE
[FEATURE REQUEST] Backup folders for Android device

### ISSUE THREAD

**Issue by rsun-thoughtworks** [owncloud/android#3126]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when new photos have been uploaded, but existing photos are ignored.

**Describe the solution you'd like**
As an user,
I would like to select many folders on my Android device, and I would like ownCloud App to automatically detect if any file does not exist on the cloud and backup them.

For example,
I have a,b,c files on my Android device under folder `media`, and I choose to backup this folder to `ThisBackup` folder on ownCloud server, and `ThisBackup` folder already have a file named `Iamhere.mp3` over there.
I added one file d to this folder on my Android device, then, this file is uploaded to owncloud server.
I remove `a` from this folder, and this `a` will be removed from owncloud server as well.
There is one exception, `Iamhere.mp3` exists before I choose to backup `media` folder on my Android device to `ThisBackup` folder, so this file should not be deleted by sync logic.

**Describe alternatives you've considered**
I can upload mannually, but if I upload one file two times, ownCloud will rename it after upload it, but what I would like it to do is to not upload the file, because it already exists on the owncloud server.

**Additional context**
No

### TASKS

- [ ] Research (if needed)
- [ ] Create branch feature/feature_name
- [ ] Development tasks
  - [ ] Implement whatever
  - [ ] ...
- [ ] Code review and apply changes requested
- [ ] Design test plan
- [ ] QA
- [ ] Merge branch feature/feature_name into master

### PR

- App
- Library (if needed)

**Comment by abelgardep** [comment]
Thanks for opening an issue with such an interesting feature. I think it would be a great addition to the ownCloud App. 

We need to move oC App to Scoped Storage and it could affect the camera uploads and features like this one. So we will keep an eye on it 👍 

It could be a duplicate of https://github.com/owncloud/android/issues/744

**Comment by jesmrec** [comment]
Similar behaviour to desktop client. This is interesting, but have some concerns, mainly in terms of security (an old and log discussion). But, as @abelgardep, we will check how it fits with scoped storage. Thanks a lot for opening!

**Comment by vinicentus** [comment]
Hi!
This is currently the only blocker for me to switch from the still PHP-based nextcloud this oCIS, which overall seems much cleaner and less cluttered.
Have you considered this now that the switch to scoped storage is completed?

The way I see this, there are two good use cases, with slightly different requirements:

**Backup**
Used for automatically backing up folders from the android phone to the cloud.
You could select which folders to back up. oCIS would then sync all files from those folders to nextcloud.
You could choose the option for nextcloud to automatically delete the files on the phone, and keep them in the cloud. Or keep the files locally on the phone.

**Sync**
Used for keeping folders in sync between the phone and the cloud.
You could select which folders to sync. oCIS would then sync all files from the phone folder to the cloud.
I think there should be a setting whether to delete files in the cloud when deleting them on the phone.
So, when the phone user manually deletes a file on the phone, oCIS would either:
Delete the file in the cloud. Or
Keep the file in the cloud. Or
Ask the user which of the above options to use.

Also, I think it would be important to allow syncing of / backup of any type of file, not just pictures and videos. Also It would be important to scan all files, not only the new pictures as is done with the current camera upload feature.

Anyways, I just wanted to continue the discussion on this.

**Comment by kavuri** [comment]
Landed on this thread googling for a backup of the files on iphone, rather than sync. Use case is, I take a bunch of photos and want them to the backed up into my owncloud server. And if I delete the file on my iphone, the copy should still exist on owncloud. Currently, only sync is supported, i.e., both file on phone and owncloud server are deleted.
Is this feature supported?

**Comment by jesmrec** [comment]
> Landed on this thread googling for a backup of the files on iphone, rather than sync. Use case is, I take a bunch of photos and want them to the backed up into my owncloud server. And if I delete the file on my iphone, the copy should still exist on owncloud. Currently, only sync is supported, i.e., both file on phone and owncloud server are deleted.
Is this feature supported?

This feature exists in the app, its name is auto-uploads. It works for all the pictures taken from the moment the feature is enabled. No matter where you delete the pictures from (ownCloud server or mobile app), the other side will keep them. You can take a look, check the feature and tell us your feelings with it.

**Comment by kavuri** [comment]
My bad, I was deleting the file from owncloud app and as per the owncloud functionality, it gets deleted from server as well. I was confused by the dual functionality of owncloud app - one of sync and another of photo viewers, camera app etc. I will be careful to use owncloud app only for sync

**Comment by jesmrec** [comment]
> My bad, I was deleting the file from owncloud app and as per the owncloud functionality, it gets deleted from server as well. I was confused by the dual functionality of owncloud app - one of sync and another of photo viewers, camera app etc. I will be careful to use owncloud app only for sync

yes, the sync works from ownCloud mobile app vs ownCloud server. My mention above about the non-delete files was over files in the device (not in the app). Be careful with that. Anyway, in the web you have a trashbin feature available to rescue deleted files.

**Comment by kavuri** [comment]
> > My bad, I was deleting the file from owncloud app and as per the owncloud functionality, it gets deleted from server as well. I was confused by the dual functionality of owncloud app - one of sync and another of photo viewers, camera app etc. I will be careful to use owncloud app only for sync
> 
> yes, the sync works from ownCloud mobile app vs ownCloud server. My mention above about the non-delete files was over files in the device (not in the app). Be careful with that. Anyway, in the web you have a trashbin feature available to rescue deleted files.

Understand. User experience wise, it would be good to have a clear demarcation of sync vs backup in the mobile app. Right now, the apps has overloaded functionality of camera, media viewer + sync. And in order to backup, we have to open the iphone photo viewer/Files to delete. So, 2 apps for 2 different kinds of functionality

**Comment by jesmrec** [comment]
@kavuri, is your device  an iPhone?, then the correct place is https://github.com/owncloud/ios-app/issues/. Even when both Android and iOS have same features, the way to implement them is different due to different system restrictions and features.
